File: 프로그래머스/lv2/12949. 행렬의 곱셈/행렬의 곱셈.js
function solution(arr1, arr2) {
    // 0으로 초기화된 2차원 배열 선언
    const result = new Array(arr1.length).fill(0).map(() => new Array(arr2[0].length).fill(0));
    
    for(let i = 0; i < result.length; i++) {
        for(let j = 0; j < result[0].length; j++) {
            let sum = 0;
            
            for(let k = 0 ; k < arr2.length; k++) {
                sum += arr1[i][k] * arr2[k][j];
            }
            
            result[i][j] = sum;
        }
    }
    
    return result;
} 
    

File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.py

def solution(arr1, arr2):
    m = len(arr1)
    k = len(arr1[0])
    n = len(arr2[0])
    
    res = [[0 for _ in range(n)] for _ in range(m)]
    
    for x in range(m):
        for y in range(n):
            s = 0
            for z in range(k):
                s += arr1[x][z] * arr2[z][y]
                
            res[x][y] = s
            
    return res
            
                
